IdentityCore/tests/MSIDURLSessionDelegateTests.m

```diff
@@ -91,12 +91,12 @@ - (void)testSessionDidReceiveChallenge_whenBlockProvided_shouldUseHandlingProvid
 - (void)testSessionTaskDidReceiveChallenge_whenNoBlockProvided_shouldPerformDefaultHandling
 {
     __auto_type delegate = [MSIDURLSessionDelegate new];
-    __auto_type session = [NSURLSession new];
-    __auto_type challendge = [NSURLAuthenticationChallenge new];
-    __auto_type task = [NSURLSessionTask new];
+    __auto_type session = [NSURLSession sharedSession];
+    __auto_type challenge = [NSURLAuthenticationChallenge new];
+    __auto_type task = [session dataTaskWithURL:[NSURL URLWithString:@"https://microsoft.com"]];
```

Comment:
Could you elaborate on the purpose of this change - creating a shared session and a task created from URL?

Reply:
For this a sharedSession was created since during travis build pipeline, build.py runs all test cases in unit test target. Since there are many test methods in the class creating a new session for each method is not good practice and was causing the tests to fail during build. Hence replaced it with sharedSession